src/solvers/dg/2d/dg.jl

```diff
@@ -1579,43 +1552,42 @@ function prolong2mortars!(dg::Dg2D, ::Val{:l2}, thread_cache)
     end
 
     # Interpolate large element face data to small interface locations
-    for v in 1:nvariables(dg)
-      if dg.l2mortars.large_sides[m] == 1 # -> large element on left side
-        if dg.l2mortars.orientations[m] == 1
-          # L2 mortars in x-direction
-          for l in 1:nnodes(dg)
-            u_large[v, l] = dg.elements.u[v, nnodes(dg), l, large_element_id]
-          end
-        else
-          # L2 mortars in y-direction
-          for l in 1:nnodes(dg)
-            u_large[v, l] = dg.elements.u[v, l, nnodes(dg), large_element_id]
-          end
-        end
-        @views dg.l2mortars.u_upper[1, v, :, m] .= dg.mortar_forward_upper * u_large[v, :]
-        @views dg.l2mortars.u_lower[1, v, :, m] .= dg.mortar_forward_lower * u_large[v, :]
-      else # large_sides[m] == 2 -> large element on right side
-        if dg.l2mortars.orientations[m] == 1
-          # L2 mortars in x-direction
-          for l in 1:nnodes(dg)
-            u_large[v, l] = dg.elements.u[v, 1, l, large_element_id]
-          end
-        else
-          # L2 mortars in y-direction
-          for l in 1:nnodes(dg)
-            u_large[v, l] = dg.elements.u[v, l, 1, large_element_id]
-          end
-        end
-        @views dg.l2mortars.u_upper[2, v, :, m] .= dg.mortar_forward_upper * u_large[v, :]
-        @views dg.l2mortars.u_lower[2, v, :, m] .= dg.mortar_forward_lower * u_large[v, :]
+    if dg.l2mortars.large_sides[m] == 1 # -> large element on left side
+      leftright = 1
+      if dg.l2mortars.orientations[m] == 1
+        # L2 mortars in x-direction
+        u_large = view(dg.elements.u, :, nnodes(dg), :, large_element_id)
+        interpolate_mortar_values!(dg, mortar_type, leftright, m, u_large)
+      else
+        # L2 mortars in y-direction
+        u_large = view(dg.elements.u, :, :, nnodes(dg), large_element_id)
+        interpolate_mortar_values!(dg, mortar_type, leftright, m, u_large)
+      end
+    else # large_sides[m] == 2 -> large element on right side
+      leftright = 2
+      if dg.l2mortars.orientations[m] == 1
+        # L2 mortars in x-direction
+        u_large = view(dg.elements.u, :, 1, :, large_element_id)
+        interpolate_mortar_values!(dg, mortar_type, leftright, m, u_large)
+      else
+        # L2 mortars in y-direction
+        u_large = view(dg.elements.u, :, :, 1, large_element_id)
+        interpolate_mortar_values!(dg, mortar_type, leftright, m, u_large)
       end
     end
   end
 end
 
+@inline function interpolate_mortar_values!(dg::Dg2D, ::Val{:l2}, leftright, m,
+                                            u_large)
+  multiply_dimensionwise!(view(dg.l2mortars.u_upper, leftright, :, :, m), dg.mortar_forward_upper, u_large)
```

Comment:
Is there a particular reason why you use `view(...)` instead of `@views`? It probably saves on computation time, but what I really like about `@views` is that it preserves the array structure in the code. What do I mean? The meaning of
```julia
@views dg.l2mortars.u_upper[leftright, :, :, m]
```
is immediatly clear to me from (eye) muscle memory, while
```julia
view(dg.l2mortars.u_upper, leftright, :, :, m)
```
needs considerable brain processing before I understand what's going on here. Thus, if there are no compelling arguments against it, I'd prefer to always use `@views` instead of `view(...)`.

Reply:
We would need something like
```julia
@views multiply_dimensionwise!(dg.l2mortars.u_upper_left[  leftright, :, :, :, m], dg.mortar_forward_lower, dg.mortar_forward_upper, u_large, fstar_tmp1)
```
instead of 
```julia
multiply_dimensionwise!(view(dg.l2mortars.u_upper_left,  leftright, :, :, :, m), dg.mortar_forward_lower, dg.mortar_forward_upper, u_large, fstar_tmp1)
```
What I don't like that much about the first approach is that it separates the important information (hey, we're using `view` here for performance) from the place where this information is needed when reading the code. Since I'm used to read `views(...)`, that's what I've used here. What do you think, shall I change everything to `@views`?

Reply:
I absolutely understand your argument about it being more elegant to have the performance-enhancing code next to where it actually makes a difference. Under other circumstances I would also agree with you that this is the way we should do it.

However, in the *particular* case of array access, which is almost universally recognized also by new users coming from, e.g., Python/numpy, C/C++ etc., the `@views` approach offers an advantage when taking a different point of view: It separates the algorithm from the performance modifications. That is, as a user who doesn't care about performance you can write your kernel as you are used to it and then just slap on `@views`, which does not change the algorithm but just makes it faster.

I think both cases can be argued and have valid arguments. For the sake of consistency, however, I'd strongly favor a single style for Trixi (which could and should be tackled in a separate PR/issue later). My personal preference is definitely `@views`, but I'd be OK with `view` if the majority favors it. What do you think, @andrewwinters5000 and @gregorgassner ?